src/scales.py

```diff
@@ -107,7 +108,54 @@ def inverted(self):
             return InverseScale.InverseTransform()
 
         def transform_non_affine(self, a):
-            return 1 / numpy.array(a)
+            # Don't spam about invalid divide by zero errors
+            with numpy.errstate(divide="ignore", invalid="ignore"):
+                return 1 / numpy.array(a)
+
+
+class CustomScaleLocator(ticker.MaxNLocator):
+    """Dynamically find tick positions on custom scales."""
+    def __init__(self, scale_type="major"):
```

Comment:
scale_type isn't my favourite here. Firstly it describes, what the ticks are so tick_type is a more suitable fit here. Secondly, there are just two types (and likely will be in mpl for the forseeable future) so a boolean like `is_major` would be more suitable imo.